scanpy/metrics/_gearys_c.py

```diff
@@ -0,0 +1,286 @@
+from typing import Optional, Union
+
+from anndata import AnnData
+from multipledispatch import dispatch
+import numba
+import numpy as np
+import pandas as pd
+from scipy import sparse
+
+
+def _choose_obs_rep(adata, *, use_raw=False, layer=None, obsm=None, obsp=None):
+    """
+    Choose array aligned with obs annotation.
+    """
+    is_layer = layer is not None
+    is_raw = use_raw is not False
+    is_obsm = obsm is not None
+    is_obsp = obsp is not None
+    choices_made = sum((is_layer, is_raw, is_obsm, is_obsp))
+    assert choices_made <= 1
+    if choices_made == 0:
+        return adata.X
+    elif is_layer:
+        return adata.layers[layer]
+    elif use_raw:
+        return adata.raw.X
+    elif is_obsm:
+        return adata.obsm[obsm]
+    elif is_obsp:
+        return adata.obsp[obsp]
+    else:
+        raise RuntimeError("You broke it. But how? Please report this.")
```

Comment:
😁 love this kind of message actually appearing.

```suggestion
        assert False, "We forgot to update this. Please report this as a bug!"
```

Generally I make these assertions, as assertions failing means the same as HTTP 500 or Rust programs panicking, i.e. “the devs fucked up. It’s not your fault but report it”